examples/02_intermediate/wikimedia.ipynb

```diff
@@ -0,0 +1,698 @@
+    "    # Create an edge in the graph with the events fetched by the adapter\n",
+    "    events = HistoricalWikiData(num_ticks=num_ticks, url=URL)\n",
+    "    # We will process the events data through a node, and only print events\n",
+    "    # happening at en.wikipedia.org\n",
+    "    en_wiki = csp.filter(events.servername == \"en.wikipedia.org\", events)\n",
+    "    # Let's pretty-print the results\n",
+    "    en_events = pretty_print(en_wiki)\n",
+    "    csp.print(\"Wiki event:\", en_events)\n",
+    "    # Add this edge as a graph output\n",
+    "    # This allows you to connect an edge as a \"graph output\", returned to the\n",
+    "    # caller from csp.run as a dictionary of key: [(datetime, value)]\n",
+    "    csp.add_graph_output(\"en_wiki\", en_wiki)\n",
+    "    print(\"End of graph building\")\n",
+    "\n",
+    "csp.run(wiki_graph, starttime=start_time, endtime=end_time)\n",
+    "print(\"Done.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c6228a65-7cd4-455c-98ce-4e3daa303e99",
+   "metadata": {},
+   "source": [
+    "---"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9f836420-e128-4f9c-afe7-a5b34309a5bd",
+   "metadata": {},
+   "source": [
+    "## Alternative workflow: using the CSP built-in CSVReader adapter"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "87ea9760-1dfc-444a-944d-d4a0915f273c",
+   "metadata": {},
+   "source": [
+    "Since our historical data can be downloaded into a csv file, we can also use the existing implementation of a [`CSVReader` adapter](https://github.com/Point72/csp/blob/main/csp/adapters/csv.py) from CSP."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b513164c-9e2f-44c9-a7c2-9a93357f4cb9",
+   "metadata": {},
+   "source": [
+    "Again, we will use a Perspective widget to display the results of our historical data. This time, the widget will immediately show the historical data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a0a16e9f-4293-45b8-a520-0caa6734b866",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datetime import datetime, timezone, timedelta\n",
+    "\n",
+    "start_time = datetime(2024, 5, 25, 0, 0, 0)\n",
+    "end_time = datetime(2024, 5, 25, 23, 0, 0, tzinfo=timezone.utc)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "555b24f7-f765-4711-90a9-46db5555a50f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import re\n",
+    "import json\n",
+    "from sseclient import SSEClient as EventSource, Event\n",
+    "from perspective import PerspectiveWidget, Table, Plugin\n",
+    "import pandas as pd\n",
+    "\n",
+    "start_date = datetime.isoformat(start_time)\n",
+    "end_date = datetime.isoformat(end_time)\n",
+    "\n",
+    "URL = f'https://stream.wikimedia.org/v2/stream/recentchange?since={start_date}'\n",
+    "\n",
+    "# Initial data\n",
+    "data = pd.DataFrame({\n",
```

Comment:
if you want to keep this CSV example here then we can leave the pandas code just for this section ( remove from the top )
though I think it may be cleaner not to have this here, may be more confusing than its worth to introduce the CSV adapter here

Reply:
No problem, I'll remove it but maybe we can save for a CSVReader how-to? We can decide later :)